src/components/TableOfContents.tsx

```diff
@@ -258,14 +77,68 @@ const TableOfContentsLink: React.FC<IPropsTableOfContentsLink> = ({
   if (isNested) {
     classes += " nested"
   }
+
+  const $dotBg = cssVar("dot-bg")
+
+  const hoverOrActiveStyle: SystemStyleObject = {
+    color: "primary",
+    _after: {
+      content: `""`,
+      background: $dotBg.reference,
+      border: "1px",
+      borderColor: "primary",
+      borderRadius: "50%",
+      boxSize: 2,
+      position: "absolute",
+      left: "-1.29rem",
+      top: "50%",
+      mt: -1,
+    },
+  }
+
   return (
-    <StyledTableOfContentsLink to={url} className={classes}>
+    <Link
+      as={GatsbyLink}
```

Comment:
Could we use our custom `Link` component for this? `src/components/Link`

Reply:
It wasn't used originally, but I will go in and add it. No problem! 😄